core/src/main/java/com/graphhopper/util/InstructionList.java

```diff
@@ -429,10 +436,52 @@ public void setPoints(PointList points) {
     }
 
     public PointList getPoints(Instruction instruction) {
+        if (instruction == null)
+            return PointList.EMPTY;
+        if (instruction.getLength() == 0)
+            // Copy does not include the last point
+            return this.points.copy(instruction.getFirst(), instruction.getLast() + 1);
         return this.points.copy(instruction.getFirst(), instruction.getLast());
```

Comment:
We should always include the last point, even if we break backward compatibility.